lib/redis.rb

```diff
@@ -2130,13 +2130,14 @@ def unwatch
 
   def pipelined
     synchronize do |client|
+      pipeline = Pipeline.new
       begin
-        original, @client = @client, Pipeline.new
+        original, @client = @client, pipeline
         yield(self)
-        original.call_pipeline(@client)
       ensure
         @client = original
       end
+      original.call_pipeline(pipeline)
```

Comment:
How would it have prevented that? Commands in the block are already executed on the above `yield`